app/stores/SchoolStore.js

```diff
@@ -139,64 +143,54 @@ function hasSchool(schoolId) {
   return typeof _schools[schoolId] !== 'undefined';
 }
 
-function _getItemForYear(school, itemListName, year) {
-  return _.find(school[itemListName], function(item) {
-    return item.begin_year <= year;
-  });
+function _getLatestYear(school) {
+  return getBeginAndEndYear(school).endYear || new Date().getFullYear();
 }
 
-function _getPrincipals(school) {
-  return _.map(school.principals, function(schoolPrincipal) {
-    const principal = PrincipalStore.getPrincipal(schoolPrincipal.id);
-    return _.assign(principal, {
-      'begin_year': schoolPrincipal.begin_year,
-      'end_year': schoolPrincipal.end_year
-    });
+function _getRelationalData(relationObjects, getter) {
+  return _.map(relationObjects, function(relationObject) {
+    return _getRelationanlObject(relationObject, getter);
   });
 }
 
-function _getSchoolByIdWrapper(func, defaultValue) {
-  return function(schoolId) {
-    defaultValue = defaultValue ? defaultValue : [];
-    const school = _schools[schoolId];
-    let newArgs = Array.prototype.slice.call(arguments, 1);
-    newArgs.unshift(school);
-    return _.isEmpty(school) ? defaultValue : func.apply(this, newArgs);
-  };
+function _getRelationanlObject(relationObject, getter) {
```

Comment:
I don't come up with any other better name, but maybe it's good to use "association" or some other term related with a junction or association table instead of relational.